cpp/src/c_api/lookup_src_dst.cpp

```diff
@@ -0,0 +1,322 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "c_api/lookup_src_dst.hpp"
+
+#include "c_api/abstract_functor.hpp"
+#include "c_api/graph.hpp"
+#include "c_api/graph_helper.hpp"
+#include "c_api/resource_handle.hpp"
+#include "c_api/utils.hpp"
+
+#include <cugraph_c/algorithms.h>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/detail/shuffle_wrappers.hpp>
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/sampling_functions.hpp>
+
+#include <raft/core/handle.hpp>
+
+#include <optional>
+
+namespace {
+
+struct build_lookup_map_functor : public cugraph::c_api::abstract_functor {
+  raft::handle_t const& handle_;
+  cugraph::c_api::cugraph_graph_t* graph_{nullptr};
+  cugraph::c_api::cugraph_lookup_container_t* result_{};
+
+  build_lookup_map_functor(::cugraph_resource_handle_t const* handle, ::cugraph_graph_t* graph)
+    : abstract_functor(),
+      handle_(*reinterpret_cast<cugraph::c_api::cugraph_resource_handle_t const*>(handle)->handle_),
+      graph_(reinterpret_cast<cugraph::c_api::cugraph_graph_t*>(graph))
+  {
+  }
+
+  template <typename vertex_t,
+            typename edge_t,
+            typename weight_t,
+            typename edge_type_type_t,
+            bool store_transposed,
+            bool multi_gpu>
+  void operator()()
+  {
+    if constexpr (!cugraph::is_candidate<vertex_t, edge_t, weight_t>::value) {
+      unsupported();
+    } else {
+      // ecg expects store_transposed == false
+      if constexpr (store_transposed) {
+        error_code_ = cugraph::c_api::
+          transpose_storage<vertex_t, edge_t, weight_t, store_transposed, multi_gpu>(
+            handle_, graph_, error_.get());
+        if (error_code_ != CUGRAPH_SUCCESS) return;
+      }
+
+      auto graph =
+        reinterpret_cast<cugraph::graph_t<vertex_t, edge_t, false, multi_gpu>*>(graph_->graph_);
+
+      auto graph_view = graph->view();
+
+      auto edge_ids = reinterpret_cast<
+        cugraph::edge_property_t<cugraph::graph_view_t<vertex_t, edge_t, false, multi_gpu>,
+                                 edge_t>*>(graph_->edge_type_);
+      auto edge_types = reinterpret_cast<
+        cugraph::edge_property_t<cugraph::graph_view_t<vertex_t, edge_t, false, multi_gpu>,
+                                 edge_type_type_t>*>(graph_->edge_type_id_type_);
+
+      auto number_map = reinterpret_cast<rmm::device_uvector<vertex_t>*>(graph_->number_map_);
+
+      auto lookup_container = new cugraph::lookup_container_t<edge_t, edge_type_type_t, vertex_t>();
+
+      std::cout << "\n\nGoing to call cugraph::build_edge_id_and_type_to_src_dst_lookup_map\n\n";
+
+      cugraph::build_edge_id_and_type_to_src_dst_lookup_map(
+        handle_, graph_view, edge_ids->view(), edge_types->view());
+
+      std::cout << "\n\nAfter calling cugraph::build_edge_id_and_type_to_src_dst_lookup_map\n\n";
+      /*
```

Comment:
Don't you need this commented out code?

Reply:
Actually all of these were removed, un-renumber added in a commit but never pushed.